mod/search.php

```diff
@@ -51,38 +51,40 @@ function search_content(App $a)
 		goaway('/home');
 	}
 
-	if ($search) {
-		$alpha = true;
+	if (strpos($search, ' ') === false) {
+		$search .= '*';
 	}
 
 	//Run our query.
-	if ($search) {
-		$search = dbesc($search . '*');
-	}
+	$search = dbesc($search);
+	$search = str_replace('%', '%%', $search);
 
-	$sql_extra = ((strlen($search)) ? " AND MATCH (`name`, `pdesc`, `homepage`, `locality`, `region`, `country-name`, `tags` )
-		AGAINST ('$search' IN BOOLEAN MODE) " : '');
+	$sql_where = "WHERE MATCH (`name`, `pdesc`, `homepage`, `locality`, `region`, `country-name`, `tags` )
+AGAINST ('$search' IN BOOLEAN MODE)
+AND NOT `censored`
+AND `available`";
 
 	if (!is_null($community)) {
-		$sql_extra .= ' AND `comm` = ' . intval($community) . ' ';
+		$sql_where .= '
+AND `comm` = ' . intval($community);
 	}
 
-	$sql_extra = str_replace('%', '%%', $sql_extra);
-
 	$total = 0;
-	$r = q("SELECT COUNT(*) AS `total` FROM `profile` WHERE `censored` = 0 AND `available` = 1 $sql_extra ");
+	$r = q("SELECT COUNT(*) AS `total`
+FROM `profile`
+$sql_where");
 	if (count($r)) {
 		$total = $r[0]['total'];
 		$a->set_pager_total($total);
 	}
 
-	if ($alpha) {
-		$order = ' ORDER BY `name` ASC ';
-	} else {
-		$order = ' ORDER BY `updated` DESC, `id` DESC ';
-	}
+	$query = "SELECT *
```

Comment:
The indentation looks wrong in the github diff. Could you have a look?

Reply:
It's tabs everywhere, what's the matter?

Reply:
It's tabs everywhere, what are you seeing wrong?

Reply:
If you're talking about the SQL query strings, it's intentional, a old habit I took when I was outputting the queries for debug and I loathed the extra white spaces in the debug display. So I started pulling the queries left to remove the tabs/spaces.